committee-code-of-conduct/election.md

```diff
@@ -0,0 +1,76 @@
+# Kubernetes Code of Conduct Committee Elections
+
+This document outlines the process, for Code of Conduct Committee (CoCC) elections.
+
+### Length of Terms
+
+* The CoCC consists of 5 members. In the first election, the top 3 voted people will be appointed for 2 year terms and the other 2 members will be appointed for a 1 year term.
+* In alternating years either 2 or 3 seats will be up for election as terms expire.
+
+### Eligibility for voting
+
+* Voting will be done by members of the steering committee.
+
+### Eligibility for candidacy
+
+* Candidate commits to putting the interests of the community above the interests of their employer for all Code of Conduct Committee activities.
+* Is generally a responsible human
+* Does not have to be part of the Kubernetes or CNCF community
+* Characteristics which would be nice in any nominations:
+  * Previous experience on an Ethics Committee or Code of Conduct Committee is appreciated
+  * Has demonstrated integrity, professionalism, and positive influence within the community is appreciated
+  * Experience with the tools which we use to communicate (Zoom, Slack, GitHub, etc.) within the Kubernetes community is appreciated
+
+### Election process
+
+The steering committee will announce a request for nominations some time in July from any member of the following:
+* steering committee
+* sig chair
+* sig lead
+
+The committee will accept nominations for at least 3 weeks. After the announced nomination
+period the steering committee will hold a private ballot election using time-limited
+[Condorcet] ranking use the [Schulze method] to break ties. The steering committee
+will publicly announce the results of the election.
+
+### Maximal representation
+
+Since we require commitment that all members of the committee put the interests
```

Comment:
It may be worth saying, "Since we require commitment... while performing all duties related to the CoCC, _or they are required to recuse themselves_" Or something along those lines.

Just a pointer to our charter where we explain that recusal is an option.

Reply:
I made a pointer to recusal. Though if a person is not able to publicly say they will put the community before their employer I don't think they belong on the committee at all.

Reply:
I'm still wary at the prospect of a mono-employer committee here.  Our other two committees explicitly call this out, why not here?

Reply:
+1 to @spiffxp - I also think we should have the same representation requirement here as we do on the steering committee. We also expect steering to put the community's interest above all as part of their responsibility to the k8s community but still have the max representation limitations as safeguards.

Reply:
Do we actually write down that steering (or any leader?) must put the interest of the community over the interest of their employer? I haven't seen it, and I've looked.

Are there other open source communities that disqualify otherwise exceptional individuals from participation based solely on employment? I'm not sure of one. Certainly not the 2 largest I can think of, kernel and OpenStack.

The only reason employer based disqualification makes any sense if if the human individuals involved are acting as proxies for their employer, instead of acting in the best interest of the community. And if they are doing so they certainly don't belong on a CoCC, or in any community leadership role in my opinion.

The very idea that because 2 people work for the same large corporation that they must think or act similarly is not rational even on it's face. Especially given the size of some of the companies in question. Personally, I don't even know the names of some of the Red Hatters who work on kubernetes and we certainly don't have shared goals or direction. We don't even have a manager in common until you get to the CEO. They are no closer to me or likely to share a thought echo chamber than anyone else in the community. And they should not be disqualified from participation if they are the best.

I think the CoCC is a good place to start fixing the idea that employment and community participation should somehow be mutually exclusive. Especially since the CoCC is the most important place to keep employment politics out of the activities.

Reply:
@eparis Yes, steering wrote this down https://github.com/kubernetes/community/blob/master/values.md#community-over-product-or-company

I still think a safeguard is necessary

Reply:
I can see cases where:
- the person reporting a CoC violation would prefer to work the case with someone who is not from the same employer as themselves, or the person they are accusing of a violation
- the person being accused of a CoC violation would prefer that the case is worked by someone who is not of the same employer as themselves, or the person accusing them of a violation

This isn't just to protect against *actual* conflicts of interest, as it's very clear that individuals in this role should be placing the interests of the community above that of their employer, but it's also to protect against any *perceived* conflict of interest, either internally or externally.

This particular issue is only a problem if most or all of the CoCC is from the same employer. Where you have a diverse committee (in many ways, but in this case, of employer), it's easy to correct and hand a particular case off to a different individual if there is ever any concerns raised that deal with a perceived conflict of interest around the employer of a CoCC member.

Reply:
There's a couple other reasons we need to have multiple companies represented:

1. If the CoC incident is *also* considered an HR incident for the employer of the accused violator, then their coworkers may be legally unable to take action inside the CoCC due to employment rules.  This could leave us without any members of the committee able to take action on an incident.

2. Just about everyone has an unconscious bias towards giving their coworkers the benefit of the doubt.  While CoCC members will consciously try to be impartial, in cases where the injured parties and the violator have different employers, having one of the other share an employer with the entire committee could result in partisan outcomes despite people's best efforts.

Reply:
Hah, I just realized that I'm proving @eparis's point that working for the same employer doesn't mean concordence.

However ... Openstack *did* have employer representation rules, they just eliminated a lot of them when the project shrank and so did the pool of employers.

Reply:
I support the points both @cblecker and @jberkus have made -- even with the most impeccable of members on the board, folks may want or need to recuse themselves due to either internal HR issues or to avoid the perception of impropriety. And, yes, in OpenStack, we had similar constraints in place on several steering committees.

How about this: maximal representation of a single employer is (quorum - 1).

So, with a 5 person committee, that's 2 people from the same company. This avoids a potential situation where, when all employees at a company need to recuse themselves, the committee is unable to make any decisions. Additionally, it necessitates that at least 3 companies are represented on the CoCC.